Client/mods/deathmatch/logic/CDownloadableResource.cpp

```diff
@@ -14,6 +14,21 @@
 CDownloadableResource::CDownloadableResource(CResource* pResource, eResourceType resourceType, const char* szName, const char* szNameShort, uint uiDownloadSize,
                                              CChecksum serverChecksum, bool bAutoDownload)
 {
+    switch (resourceType)
+    {
+        case eResourceType::RESOURCE_FILE_TYPE_SCRIPT:
+        case eResourceType::RESOURCE_FILE_TYPE_CLIENT_SCRIPT:
+            m_resourceCategoryType = eResourceCategory::SCRIPTS;
+        case eResourceType::RESOURCE_FILE_TYPE_MAP:
+            m_resourceCategoryType = eResourceCategory::MAPS;
+        case eResourceType::RESOURCE_FILE_TYPE_CONFIG:
+        case eResourceType::RESOURCE_FILE_TYPE_CLIENT_CONFIG:
+            m_resourceCategoryType = eResourceCategory::CONFIGS;
+        case eResourceType::RESOURCE_FILE_TYPE_CLIENT_FILE:
+        case eResourceType::RESOURCE_FILE_TYPE_HTML:
+            m_resourceCategoryType = eResourceCategory::FILES;
+    }
+
```

Comment:
You can filter it in `CLuaResourceDefs::GetResourceFiles`. You don't need `GetResourceCategoryType`

Reply:
Its better this way

Reply:
Hmm. You can use original types.  I think `client-scripts` and `server-scripts` have some sense for developers. So you shouldn't map `eResourceType` into custom enum for one Lua function

Reply:
I'd have to do string manipulation in both functions just to get values.
Its better that way

Reply:
What do you mean?

Reply:
ArgumentParser handles enum convertion for me. I'd have to accept certain string values and filter them. Its too much hassle.
Its better to create an enum. Everything will be taken care of for me.

Reply:
What is wrong with `eResourceType` ?

Reply:
too much noise

Reply:
It looks like using existing enum is much better than creating a new one with mapping from `eResourceType` with adding  boilerplate code in client and server.

> too much noise
What do you mean?

Reply:
> What do you mean?

Exactly what I said. Its easier to have 3 categories: scripts, maps, files instead of: client/server scripts, client/server files, (client/server?) maps, etc.

Reply:
I see no problem here. It's much useful.